test/unit/prof_hook.c

```diff
@@ -0,0 +1,56 @@
+#include "test/jemalloc_test.h"
+
+bool mock_bt_hook_called = false;
+
+void
+mock_bt_hook(void **vec, unsigned *len, unsigned max_len) {
+	*len = max_len;
+	for (unsigned i = 0; i < max_len; ++i) {
+		vec[i] = (void *)((uintptr_t)i);
+	}
+	mock_bt_hook_called = true;
```

Comment:
Let's test one more case: in the mock_hook, invoke the default hook, and then expand the `vec` returned from the default hook by a few. This should be a fair use case.